modules/calcite/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -148,19 +148,29 @@ SqlCreate SqlCreateTable(Span s, boolean replace) :
     final SqlIdentifier id;
     final SqlNodeList columnList;
     final SqlNodeList optionList;
+    final SqlNode query;
 }
 {
     <TABLE>
     ifNotExists = IfNotExistsOpt()
     id = CompoundIdentifier()
-    columnList = TableElementList()
+    (
+        columnList = TableElementList()
+    |
+        { columnList = null; }
+    )
+    (
+        <AS> query = OrderedQueryOrExpr(ExprContext.ACCEPT_QUERY)
+    |
+        { query = null; }
+    )
     (
         <WITH> { s.add(this); } optionList = CreateTableOptionList()
     |
         { optionList = null; }
     )
     {
```

Comment:
Please extend `org.apache.ignite.internal.processors.query.calcite.sql.SqlDdlParserTest` with tests to verify the new clause format

Reply:
New test added